### ISSUE TITLE
global_identifier does not work when using Softlayer driver

### ISSUE THREAD

**Issue by giany** [saltstack/salt#35481]
### Description of Issue/Question

Cannot create Softlayer instances based on a custom image.

```
[root@stage salt]# salt-cloud     -P --profile debian-8-1gb-1cpu  testserver
testserver:
    ----------
    Provisioning:
        VM being provisioned in parallel. PID: 10755
[ERROR   ] The required 'image' configuration setting is missing from the 'debian-8-1gb-1cpu' profile, which is configured under the 'soft' alias.
```
### Steps to Reproduce Issue

/etc/salt/cloud.providers.d/softlayer.conf contains:

```
soft:
 user: mysluser
 apikey: 'apikey'
 driver: softlayer
```

/etc/salt/cloud.profiles.d/soft-profile.conf contains:

```
debian-8-1gb-1cpu:
  provider: soft
  #image: DEBIAN_8_64
  cpu_number: 1
  ram: 1024
  disk_size: 25
  local_disk: True
  hourly_billing: True
  domain: myhost.com
  location: ams03
  max_net_speed: 1000
  public_vlan: ams03.fcr01a.1234
  private_vlan: ams03.bcr01a.1235
  private_network: False
  private_ssh: False
  # May be used _instead_of_ image
  ssh_agent: True
  wait_for_fun_timeout: 1200
  global_identifier: <globalIdentifier taken from salt-cloud -f list_custom_images soft>
```
### Versions Report

```
[root@stage salt]# salt --versions-report
Salt Version:
           Salt: 2016.3.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.28.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core
```

Documentation at http://salt-cloud.readthedocs.io/en/latest/topics/softlayer.html#global-identifier clearly says that **image** can be skipped if you have global_identifier.

I think problem is in salt/config/**init**.py on line:

```
non_image_drivers = ['vmware', 'nova']
```

If I add there softlayer it behaves properly.

**Comment by Ch3LL** [comment]
@giany definitely looks like we need to get this added to `non_image_drivers` although I would imagine we would want to create some more logic so that its not always added to the `non_image_drivers` similar to teh code for linode [here](https://github.com/saltstack/salt/blob/develop/salt/config/__init__.py#L2886) in case a user does want to point to an image

**Comment by rallytime** [comment]
Fixed with #41551
